WordPress/Classes/ViewRelated/Post/PostCardStatusViewModel.swift

```diff
@@ -56,15 +56,30 @@ class PostCardStatusViewModel: NSObject {
         if MediaCoordinator.shared.isUploadingMedia(for: post) {
             return NSLocalizedString("Uploading media...", comment: "Message displayed on a post's card while the post is uploading media")
         } else if post.isFailed {
-            if canCancelAutoUpload {
-                return StatusMessages.postWillBePublished
+            let defaultFailedMessage = StatusMessages.uploadFailed
+            guard let postStatus = post.status else {
+                return defaultFailedMessage
             }
 
             if post.wasAutoUploadCancelled {
                 return StatusMessages.localChanges
             }
 
-            return StatusMessages.uploadFailed
+            if autoUploadInteractor.autoUploadAction(for: post) != .upload {
```

Comment:
Thank you for investigating, @leandroalonso. The `shouldAttemptAutoUpload` returns `false`, that's why the action is `.autoSave` instead of `.upload`. The `shouldAttemptAutoUpload` is initially set to `true` here: 

https://github.com/wordpress-mobile/WordPress-iOS/blob/2c6380a68b583dafe3eba8bb7cccaa69790bbf5b/WordPress/Classes/ViewRelated/Post/PostEditor%2BPublish.swift#L87

However, something in our code changed the `status` to `.failed` which resulted in the hash changing, and ultimately the `shouldAttemptAutoUpload` return value.

https://github.com/wordpress-mobile/WordPress-iOS/blob/2c6380a68b583dafe3eba8bb7cccaa69790bbf5b/WordPress/Classes/Models/AbstractPost%2BHashHelpers.m#L7-L25

This is a case we probably did not find during #12252. This is a separate issue though since it happens in the master branch too. I created https://github.com/wordpress-mobile/WordPress-iOS/issues/12498 for it.